File: kotlin/difference-of-squares/src/main/kotlin/DifferenceOfSquares.kt
class Squares (val number: Int) {

    fun sumOfSquares() : Int {
        return (this.number * (this.number + 1) * (2 * this.number + 1)) / 6
    }

    fun squareOfSum() : Int {
        val sum = (this.number * (this.number + 1)) / 2
        return sum * sum
    }

    fun difference() : Int{
        return this.squareOfSum() - this.sumOfSquares()
    }
}

File: javascript/difference-of-squares/difference-of-squares.js
export class Squares {
    constructor(number) {
        this.number = number;
    }

    get sumOfSquares() {
        return (this.number * (this.number + 1) * (2 * this.number + 1)) / 6;
    }

    get squareOfSum() {
        let sumOfNumbers = (this.number * (this.number + 1)) / 2;
        return sumOfNumbers * sumOfNumbers;
    }

    get difference() {
        return this.squareOfSum - this.sumOfSquares;
    }
}
